oauth2/src/main/java/com/linecorp/armeria/common/auth/oauth2/UnsupportedMediaTypeException.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.auth.oauth2;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.HttpHeaderNames;
+
+/**
+ * A response {@code Content-Type} header does not match the expected type.
+ */
+public class UnsupportedMediaTypeException extends RuntimeException {
```

Comment:
- How about introducing a new parent exception such as `OAuth2Exception`? `TokenRequestException` could extend it as well.
- The exception name sounds too generic. Should we add `OAuth2` in its name? e.g. `UnsupportedOAuth2MediaTypeException`?

Reply:
`UnsupportedResponseException` and `UnsupportedMediaTypeException` are indeed generic exceptions. I wish they were placed somewhere inside the core internal. These are not related to OAuth2 or token processing.

Reply:
How about simply replacing `UnsupportedResponse/MediaTypeException` with an `InvalidResponseException` which is a sister of `InvalidRequestException`? That would make it clear that the failure occurred due to a bad OAuth2 response.

Reply:
Introduced `InvalidResponseException` as a parent of `UnsupportedResponse/MediaTypeException`.